server/server.go

```diff
@@ -1711,14 +1753,35 @@ func (server *BgpServer) deleteNeighbor(c *config.Neighbor, code, subcode uint8)
 	return nil
 }
 
+func (s *BgpServer) DeletePeerGroup(c *config.PeerGroup) error {
+	return s.mgmtOperation(func() error {
+		return s.deletePeerGroup(c)
+	}, true)
+}
+
 func (s *BgpServer) DeleteNeighbor(c *config.Neighbor) error {
 	return s.mgmtOperation(func() error {
 		return s.deleteNeighbor(c, bgp.BGP_ERROR_CEASE, bgp.BGP_ERROR_SUB_PEER_DECONFIGURED)
 	}, true)
 }
 
+func (s *BgpServer) UpdatePeerGroup(pg *config.PeerGroup) error {
```

Comment:
I think it is better to move this logic in `(*BgpServer).UpdatePeerGroup` to support `peer-group` feature for native go library users. Also, when we support `peer-group` configuration via gRPC API, it would be easier to implement it if the logic only remains in `BgpServer`.